src/ukify/ukify.py

```diff
@@ -711,6 +743,112 @@ def make_uki(opts):
     print(f"Wrote {'signed' if sign_args_present else 'unsigned'} {opts.output}")
 
 
+ONE_DAY = datetime.timedelta(1, 0, 0)
+
+
+@contextlib.contextmanager
+def temporary_umask(mask: int):
+    # Drop <mask> bits from umask
+    old = os.umask(0)
+    os.umask(old | mask)
+    try:
+        yield
+    finally:
+        os.umask(old)
+
+
+def generate_key_cert_pair(
+        common_name: str,
+        keylength: int = 2048,
+        valid_days: int = 365 * 10,  # TODO: can we drop the expiration date?
```

Comment:
`cryptography` seems to to want one and error out with a `ValueError` if one is not given - I don't know if the standard requires one. Also not knowing the standard, could it be an issue that ten years in the future will soon be after 19 January 2038? 

Anyway, the key validity would probably be something that people might want to set as an option and in mkosi we have an option for that. (We also have one for the keylength, but in light of the below comment that's probably unnecessary)

Reply:
Done now.